Tests/Realm.Tests/Sync/SessionTests.cs

```diff
@@ -39,6 +42,13 @@ public class SessionTests : SyncTestBase
     {
         private readonly ConcurrentQueue<EventHandler<ErrorEventArgs>> _sessionErrorHandlers = new();
 
+        private static IEnumerable<IClientResetHandler> _allClientResetHandlers()
+        {
+            yield return new ClientResetHandler<DiscardLocalResetHandler>();
+            yield return new ClientResetHandler<AutomaticRecoveryHandler>(AutomaticRecoveryHandler.Fallback.DiscardLocal);
+            yield return new ClientResetHandler<AutomaticRecoveryHandler>(AutomaticRecoveryHandler.Fallback.None);
+        }
```

Comment:
This seems a bit overengineered and will not work for Unity. When passing arguments to `TestCaseSource`, you need to have them be `object[]`. My suggestion would be to do something like:

```csharp
public static readonly object[] AllClientResetHandlers = new[]
{
    new object[] { typeof(DiscardLocalHandler), (AutomaticRecoveryHandler.Fallback?)null },
    new object[] { typeof(AutomaticRecoveryHandler), AutomaticRecoveryHandler.Fallback.DiscardLocal },
    new object[] { typeof(AutomaticRecoveryHandler), AutomaticRecoveryHandler.Fallback.None },
};

[TestCaseSource(nameof(AllClientResetHandlers))]
public void MyTest(Type handlerType, AutomaticRecoveryHandler.Fallback? fallback)
{

}

private static ClientResetHandlerBase GetClientResetHandler(Type type, 
    AutomaticRecoveryHandler.Fallback? fallback,
    BeforeResetCallback beforeCb = null,
    AfterResetCallback afterCb = null,
    ClientResetCallback manualCb = null)
{
    var handler = fallback.HasValue ? Activator.CreateInstance(type, new[] { fallback.Value }) : Activator.CreateInstance(type);

    if (beforeCb != null)
    {
        type.GetProperty(nameof(DiscardLocalHandler. OnBeforeReset)).SetValue(handler, beforeCb);
    }

    // rest of callbacks
}
```

Reply:
Is there a specific reason for `AllClientResetHandlers` to be public?